### PR TITLE
fix issues with HMD lookout 2D, and cleanup

### PR DESCRIPTION
**Pull request by ZappoMan**
- fixes a bug in `HMDScriptingInterface::getHUDLookAtPosition2D()` that was returning incorrect values for the 2D position of what you were looking at.
- removed a bunch of dead code

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b76cf484df7ca686feeb549d3119e5a185ffc96 available at: https://gist.github.com/af9bd0d2d3967b0771b3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6834.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6834.dmg

**Comment by ZappoMan** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b76cf484df7ca686feeb549d3119e5a185ffc96 available at: https://gist.github.com/af9bd0d2d3967b0771b3<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/pr-interface-win64-PR-6834.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6834.dmg

File: interface/src/scripting/HMDScriptingInterface.cpp [hunk 1]
```diff
@@ -27,10 +27,9 @@ QScriptValue HMDScriptingInterface::getHUDLookAtPosition2D(QScriptContext* conte
         MyAvatar* myAvatar = DependencyManager::get<AvatarManager>()->getMyAvatar();
         glm::vec3 sphereCenter = myAvatar->getDefaultEyePosition();
         glm::vec3 direction = glm::inverse(myAvatar->getOrientation()) * (hudIntersection - sphereCenter);
-        glm::quat rotation = ::rotationBetween(glm::vec3(0.0f, 0.0f, -1.0f), direction);
-        glm::vec3 eulers = ::safeEulerAngles(rotation);
-        return qScriptValueFromValue<glm::vec2>(engine, qApp->getApplicationCompositor()
-            .sphericalToOverlay(glm::vec2(eulers.y, -eulers.x)));
+        glm::vec2 polar = glm::vec2(glm::atan(direction.x, -direction.z), glm::asin(direction.y)) * -1.0f;
+        auto overlayPos = qApp->getApplicationCompositor().sphericalToOverlay(polar);
```
Comment (jherico) [inline interface/src/scripting/HMDScriptingInterface.cpp]: I'd suggest that we make an `ApplicationCompositor::directionToOverlay(const glm::vec3& direction)` that combines incorporates this polar conversion.

Comment (ZappoMan) [inline interface/src/scripting/HMDScriptingInterface.cpp]: I do plan to do that... but I wanted to do it in the context of reworking some other stuff (in the handControllerGrab usage of cursor movement)... so I'm hoping to merge this as is, and improve in my next PR. ok?

**Pull request merged**